File: backend/voting-app/src/main/java/com/tu/votingapp/dto/request/DocumentRequestDTO.java
package com.tu.votingapp.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DocumentRequestDTO {
    @NotBlank(message = "Document number is required")
    private String number;

    @NotNull(message = "Valid from date is required")
    private LocalDate validFrom;

    @NotNull(message = "Valid to date is required")
    private LocalDate validTo;

    @NotBlank(message = "Issuer is required")
    private String issuer;

    @NotNull(message = "Gender is required")
    private Integer gender;

    @NotNull(message = "Date of birth is required")
    private LocalDate dateOfBirth;

    @NotBlank(message = "Permanent address is required")
    private String permanentAddress;
}

File: frontend/android/app/src/main/java/com/example/android/dto/request/DocumentRequestDTO.kt
package com.example.android.dto.request

import kotlinx.serialization.Serializable

@Serializable
data class DocumentRequestDTO(
    val number: String,
    val validFrom: String,
    val validTo: String,
    val issuer: String,
    val gender: Int,
    val dateOfBirth: String,
    val permanentAddress: String
)